addons/html_editor/static/src/main/list/list_plugin.js

```diff
@@ -525,43 +525,43 @@ export class ListPlugin extends Plugin {
         cursors.restore();
     }
 
+    /**
+     * @param {HTMLLIElement} li
+     */
     outdentTopLevelLI(li) {
         const cursors = this.dependencies.selection.preserveSelection();
         const ul = li.parentNode;
         const dir = ul.getAttribute("dir");
-        let p;
-        let toMove = li.lastChild;
-        while (toMove) {
-            if (isBlock(toMove)) {
-                if (p && isVisible(p)) {
-                    cursors.update(callbacksForCursorUpdate.after(ul, p));
-                    ul.after(p);
-                }
-                p = undefined;
-                cursors.update(callbacksForCursorUpdate.after(ul, toMove));
-                ul.after(toMove);
-            } else {
-                p = p || this.document.createElement("P");
-                if (dir) {
-                    p.setAttribute("dir", dir);
-                    p.style.setProperty("text-align", ul.style.getPropertyValue("text-align"));
-                }
-                cursors.update(callbacksForCursorUpdate.prepend(p, toMove));
-                p.prepend(toMove);
+        const textAlign = ul.style.getPropertyValue("text-align");
+        wrapInlinesInBlocks(li, cursors);
+        if (!li.hasChildNodes()) {
+            // Outdenting an empty LI produces an empty P
+            const p = this.document.createElement("p");
+            p.append(this.document.createElement("br"));
+            li.append(p);
+            cursors.remapNode(li, p);
+        }
+        // Move LI's children to after UL
+        for (const block of childNodes(li).reverse()) {
+            if (dir && !block.getAttribute("dir")) {
+                block.setAttribute("dir", dir);
             }
-            toMove = li.lastChild;
-        }
-        if (p && isVisible(p)) {
-            cursors.update(callbacksForCursorUpdate.after(ul, p));
-            ul.after(p);
+            if (textAlign && !block.style.getPropertyValue("text-align")) {
+                block.style.setProperty("text-align", textAlign);
+            }
+            cursors.update(callbacksForCursorUpdate.after(ul, block));
+            ul.after(block);
         }
+        // Remove LI
         cursors.update(callbacksForCursorUpdate.remove(li));
         li.remove();
+        // Remove UL if left empty
         if (!ul.firstElementChild) {
             cursors.update(callbacksForCursorUpdate.remove(ul));
             ul.remove();
         }
         cursors.restore();
+        return;
```

Comment:
Is there a point to this `return`? Since it is useless, I'm afraid that there might be a mistake, like some missing return value or something... ^^'